compiler/rustc_traits/src/normalize_erasing_regions.rs

```diff
@@ -40,7 +41,13 @@ fn normalize_generic_arg_after_erasing_regions<'tcx>(
                 debug_assert!(!erased.needs_infer(), "{:?}", erased);
                 erased
             }
-            Err(NoSolution) => bug!("could not fully normalize `{:?}`", value),
```

Comment:
this `bug` is quite valuable and helped us catch some bug with const generics, so I would really like to keep it that way.

We pretty much only emit a bug here if the type is not well formed, and I think `normalize_erasing_regions` should not be used on types that could still be well formed.

Reply:
I think having `delay_span_bug` here should be fine. This mimics `bug` with `-Ztreat-err-as-bug` right?

Reply:
We get there with const evaluatable bounds when emitting errors, but these errors should be silent when encountered during selection, so just always causing an ICE here is preferable for me.

We shouldn't hit this, even after having a compilation error. Having a `delay_span_bug` here hides other bugs in the compiler.